storage/innobase/trx/trx0rseg.cc

```diff
@@ -662,22 +662,26 @@ dberr_t trx_rseg_array_init()
 	}
 
 #ifdef WITH_WSREP
-	if (!wsrep_sys_xid.is_null()) {
+	if (srv_operation == SRV_OPERATION_NORMAL && !wsrep_sys_xid.is_null()) {
 		/* Upgrade from a version prior to 10.3.5,
 		where WSREP XID was stored in TRX_SYS page.
 		If no rollback segment has a WSREP XID set,
 		we must copy the XID found in TRX_SYS page
 		to rollback segments. */
 		mtr.start();
+		const buf_block_t *sys = trx_sysf_get(&mtr);
+		if (memcmp(TRX_SYS + TRX_SYS_WSREP_XID_INFO + sys->frame,
+		           field_ref_zero, TRX_SYS_WSREP_XID_LEN) != 0) {
```

Comment:
In the body of `if (rseg_id == 0)` earlier, we could check if the `sys` page needs to be updated. In that way we could avoid an extra buffer pool lookup as well as the indentation changes here.
```c++
	if (wsrep_xid_in_sys && srv_operation == SRV_OPERATION_NORMAL
	    && !wsrep_sys_xid.is_null()) {
```